packages/contracts/contracts/examples/UnirepVoting.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+import {Unirep} from '.././Unirep.sol';
+
+// Uncomment this line to use console.log
+// import 'hardhat/console.sol';
+import {ReputationVerifierHelper} from '../verifierHelpers/ReputationVerifierHelper.sol';
+
+contract UnirepVoting {
+    Unirep public unirep;
+    ReputationVerifierHelper public helper;
+    uint8 public numOptions;
+    uint8 public numVoters;
+    int[] public scores;
+    mapping(uint256 => uint256) public voted;
+
+    constructor(
+        Unirep _unirep,
+        ReputationVerifierHelper _helper,
+        uint8 _numOptions,
+        uint8 _numVoters,
+        uint48 _epochLength
+    ) {
+        // set unirep address
+        unirep = _unirep;
+
+        // set verifier address
+        helper = _helper;
+
+        numOptions = _numOptions;
+        numVoters = _numVoters;
+
+        scores = new int[](numOptions);
+
+        unirep.attesterSignUp(_epochLength);
+    }
+
+    // sign up users in this app
+    function userSignUp(
+        uint256[] calldata publicSignals,
+        uint256[8] calldata proof
+    ) public {
+        unirep.userSignUp(publicSignals, proof);
+    }
+
+    function vote(
+        uint256 option,
+        uint256[] calldata publicSignals,
+        uint256[8] calldata proof
+    ) public {
+        require(option < numOptions, 'voted option out of range');
+
+        ReputationVerifierHelper.ReputationSignals
+            memory signals = decodeAndVerify(publicSignals, proof);
+
+        require(voted[signals.epochKey] < 1, 'Already voted');
+
+        if (signals.maxRep > 0) {
+            voted[signals.epochKey] += 1;
+        }
+        scores[option] += 1;
+
+        uint48 epoch = unirep.attesterCurrentEpoch(uint160(address(this)));
+        require(epoch == 0, 'not voting epoch');
+    }
+
+    // decode and verify the proofs
+    // fails or returns proof signals
+    function decodeAndVerify(
+        uint256[] calldata publicSignals,
+        uint256[8] calldata proof
+    ) public view returns (ReputationVerifierHelper.ReputationSignals memory) {
+        return helper.verifyAndCheck(publicSignals, proof);
+    }
```

Comment:
It can just call the `helper.verifyAndCheck(publicSignals, proof);`
we don't need the extra function?